source/php/Modules/FrontendForm/FormState.php

```diff
@@ -2,22 +2,22 @@
 
 namespace EventManager\Modules\FrontendForm;
 
+use WpService\Contracts\GetQueryVar;
+
 class FormState
 {
     public bool $isValidStep;
-    public int $currentStep;
     public ?int $nextStep;
     public ?int $previousStep;
     public int $totalSteps;
     public int $percentageCompleted;
+    public bool $hasStoredData;
+    public bool $isCompleted;
 
-    public function __construct(array $steps, string $queryParam = '')
+    public function __construct(array $steps, public int $currentStep)
```

Comment:
There are quite a lot assignments going on in the constructor. Would it not suffice to expose the methods instead of the variables?